docs/scos/dev/feature-walkthroughs/202204.0/configurable-product-feature-walkthrough/configurable-product-flow-chart.md

```diff
@@ -0,0 +1,198 @@
+
+## Product configuration data flow on the product details page
+
+When configuration starts on Yves, from the product details page (PDP), the product configuration data flow consists of the following phases.
+
+### Phase 1
+
+- We are on a PDP—the page with the product configuration.
+- The product configuration can be already complete or use not complete pre-configuration defined by the shop owner.
+- The product configuration can be taken from two sources:
+  - A session for complete configuration.
+  - Storage (Redis) for the pre-configuration.
+
+The following table shows the configuration data that is stored in the Session and Storage.
+
+| PARAMETER | VALUE | COMMENTS  |
+|---|---|---|
+| `ProductConfigurationInstance.isComplete` | `true` or `false` | Sensitive data. |
+| `ProductConfigurationInstance.displayDescription` | Some text of JSON blob—fore example, `["color"=>"red", "weight"=> 100]` |   |
+| `ProductConfigurationInstance.configuratorKey` | `dateTime` |   |
+| `ProductConfigurationInstance.configuration` |  `["color"=>"red", "weight"=> 100]` | Sensitive data. |
+
+We will generate a URL that points to the gateway page with the following parameters:
+
+| PARAMETER | VALUE | COMMENT |
+|---|---|---|
+| `sourceType` | `pdp`, `cart-page` | Defines the page type where the configurator request is started. Resolves the `backUrl`, when the configurator response is processed. |
+| `SKU` | `some_sku` |   |
+| `Quantity` | `2` |   |
+
+The following table contains request parameters which the plugin adds to the request:
+
+| PARAMETER | VALUE |
+|---|---|
+| `idCustomer` | `DE-1` |
+| `localeName` | `de_DE` |
+| `storeName` | `DE` |
+| `currencyCode` | `EUR` |
+| `priceMode` | `NET_MODE` or `GROSS_MODE` |
+| `backUrl` | `https://SOME_URL` |
+
+### Phase 2
+
+The customer clicks the configuration button, and the request is redirected to the gateway page with a given product configuration using a `GET` method.
+
+### Phase 3
+
+1. The configuration data is read from the session or storage for the given SKU.
+2. A plugin that handles the request is selected based on the configurator type.
+3. The plugin expands the request with additional necessary data:
+  - store, locale, currency, customer, price mode.
+  - `backUrl`—based on the referer header.
+4. The plugin generates the URL that points to the configurator page together with all the necessary data.
+
+### Phase 4
+
+1. The request comes from the gateway to the configurator `index.php` (phase 4.1).
+2. The configurator handles the request by creating a new session and saving the request's data there (phase 4.2).
+3. The configurator responds with a redirect URL that contains what session ID needs to be picked up (phase 4.2).
+4. The request comes from the gateway to the configurator `index.php` to pick up a session (phase 4.3).
+5. Redirects the customer to the configurator page by a GET request using a secured connection.
```

Comment:
Add the subject. Is it "request" or anything else?
```suggestion
5. The request redirects the customer to the configurator page by a GET request using a secured connection.
```

Reply:
"Response"